### PR TITLE
Require Java 11 to run the extension

### PR DESCRIPTION
**Pull request by fbricon**
follow up on https://github.com/eclipse/eclipse.jdt.ls/pull/1509

### CODE REVIEW

File: src/requirements.ts [hunk 1]
```diff
@@ -85,29 +85,9 @@ function checkJavaVersion(javaHome: string): Promise<number> {
...
-                                if (selection === dontShowAgain) {
-                                    getJavaConfiguration().update(section, false, ConfigurationTarget.Global);
-                                }
-                                if (selection === getJDK) {
-                                    commands.executeCommand(Commands.OPEN_BROWSER, Uri.parse(getJdkUrl()));
-                                }
-                            });
-                    }
-                }
-                resolve(javaVersion);
```
Comment (snjeza) [inline src/requirements.ts:109]: You should add this line.

Comment (fbricon) [reply src/requirements.ts:109]: Yes obviously :man_facepalming:. 
I wonder why the extension did work in regular vscode though. I must have seriously fucked up

File: demolixion2/src/main/java/foo/bar/demo/App.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+/**
+ * Hello world!
+ *
+ */
+public class App 
+{
+    public static void main( String[] args )
+    {
+        System.out.println( "Hello World!"
```
Comment (Eskibear) [inline demolixion2/src/main/java/foo/bar/demo/App.java:11]: just wondering what's this file for?

Comment (fbricon) [reply demolixion2/src/main/java/foo/bar/demo/App.java]: good catch. Removed it

**Comment by frankxls** [comment]
I'm not able to run some projects in Java 8 anymore. I guess is because of this "enhancement", however I still need to run such version. If not, there's no point in using VSCode and this extension. Thanks anyway.

**Comment by fbricon** [comment]
https://github.com/redhat-developer/vscode-java/wiki/JDK-Requirements#do-i-need-to-migrate-my-projects-to-java-11

**Comment by qq849546387** [comment]
I have setting "java.configuration.runtimes", but no effect, so I hava to change it to v0.64.1.

**Comment by tigerinus** [comment]
> I have setting "java.configuration.runtimes", but no effect, so I hava to change it to v0.64.1.

How did you revert to previous version? I could only find v0.61 at https://github.com/redhat-developer/vscode-java/releases

**Comment by tigerinus** [comment]
@frankxls @qq849546387 - from the description it sounds like JDK11 is only needed for this extension to run. It should not impact your JDK8 based development work if you setup is correct.

JDK8 is mandatory to my work too. I do not need to change either my `JDK_HOME` or `JAVA_HOME` pointing to JDK8/JRE8. My setup below works fine for me:

```json
    "java.home": "/opt/jdk11",
    "java.configuration.runtimes": [
        {
            "name": "JavaSE-1.8",
            "path": "/opt/jdk8",
            "default": true
        }
    ]
```

On the other hand for owners of this extension, maybe you should consider adding a new setting like `jdt.java.home` (in the same way like `xml.java.home` from XML extension and `metals.javaHome` from Scala extension), instead of using `java.home`.

**Comment by gajduk** [comment]
enhancement - Is that the new word for breaking change?

**Pull request merged**